dcm2niix_gear/utils/metadata.py

```diff
@@ -219,43 +219,50 @@ def capture(
+            substr = file.split(stem)
+            # If the "root" name of sidecar is not a substring of image file
+            #   We don't care about this image file.
+            if len(substr) < 2:
+                continue
+            # Get "root" and what's "left" over after splitting of "root" name
+            #   from sidecar
+            _, left = substr
 
-            if Path(sidecar).stem in Path(file).stem:
-
-                file_type = "".join(Path(file).suffixes)
-
-                if retain_nifti:
+            if retain_nifti:
 
-                    # NIfTI
-                    if file_type in [".nii.gz", ".nii"]:
-                        filedata = create_file_metadata(
-                            file, "nifti", classification, metadata, modality
-                        )
-                        capture_metadata.append(filedata)
+                # NIfTI
+                if left in [".nii.gz", ".nii"]:
+                    filedata = create_file_metadata(
+                        file, "nifti", classification, metadata, modality
+                    )
+                    capture_metadata.append(filedata)
 
-                    # bval
-                    if file_type in [".bval"]:
-                        filedata = create_file_metadata(
-                            file, "bval", classification, metadata, modality
-                        )
-                        capture_metadata.append(filedata)
+                # bval
+                if left in [".bval"]:
+                    filedata = create_file_metadata(
+                        file, "bval", classification, metadata, modality
+                    )
+                    capture_metadata.append(filedata)
 
-                    # bvec
-                    if file_type in [".bvec"]:
-                        filedata = create_file_metadata(
-                            file, "bvec", classification, metadata, modality
-                        )
-                        capture_metadata.append(filedata)
+                # bvec
+                if left in [".bvec"]:
+                    filedata = create_file_metadata(
+                        file, "bvec", classification, metadata, modality
+                    )
+                    capture_metadata.append(filedata)
 
-                if output_nrrd:
+            if output_nrrd:
 
-                    # NRRD
-                    if file_type in [".raw.gz", ".nhdr", ".nrrd"]:
-                        filedata = create_file_metadata(
-                            file, "nrrd", classification, metadata, modality
-                        )
-                        capture_metadata.append(filedata)
+                # NRRD
+                if left in [".raw.gz", ".nhdr", ".nrrd"]:
+                    filedata = create_file_metadata(
+                        file, "nrrd", classification, metadata, modality
+                    )
+                    capture_metadata.append(filedata)
 
```

Comment:
How hard would it be to split the above section into a separate function and add tests for it?